templates/base.html

```diff
@@ -2,20 +2,76 @@
 
 <!DOCTYPE html>
 <html lang="en-gb">
-<meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
 
-<head>
+  <head>
     <title>{% block title %} {% endblock %}| Share Spots</title>
     <!-- Latest compiled and minified CSS -->
-</head>
-
-<body>
+  </head>
 
+  <body>
     {% block layout %} {% endblock %}
 
     <script src="{% static 'main.js' %}" type="module"></script>
-    <link href='https://fonts.googleapis.com/css?family=Monoton|Roboto' rel='stylesheet' type='text/css'>
-    <link href='https://fonts.googleapis.com/css?family=Raleway:300,700,500' rel='stylesheet' type='text/css'>
-</body>
+    <link
+      href="https://fonts.googleapis.com/css?family=Monoton|Roboto"
+      rel="stylesheet"
+      type="text/css"
+    />
+    <link
+      href="https://fonts.googleapis.com/css?family=Raleway:300,700,500"
+      rel="stylesheet"
+      type="text/css"
+    />
+
+    {% if venue_detail.latitude %}
+
+    <!-- Map styles and JavaScript  -->
+
+    <style>
```

Comment:
we can make the map as a component and apply the styles there `src\styles\components`

Reply:
I tried but for Leaflet.js to work, the CSS and JavaScript need to load in a particular order otherwise the map won’t show, or the map isn’t displayed properly.

“Leaflet JavaScript file **after** Leaflet’s CSS” - https://leafletjs.com/examples/quick-start/

Perhaps you could try refactoring it?

However, I couldn’t manage it, which is why I end up doing the code like this.

I suspect we need to change the way we are bundling CSS, it looks like we are doing something weird where CSS is being bundled into the JavaScript file (not bundled into a CSS file in the <head>) and so the loading order isn’t right for Leaft.js
https://github.com/PolyglotDevsLondon/sharespots/blob/00307b526e7c27951bdb3851d8a6e869fefe78bf/webpack.config.js#L24

Think we need to change this in WebPack if we want to put the map in a component. Perhaps we should creat a Trello ticket to replace css-loader and then refactor the Map and Leaflet.js code?

Reply:
hmmm, need to have a look at this one!